discovery/scaleway/instance.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package scaleway
+
+import (
+	"context"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/common/model"
+	"github.com/scaleway/scaleway-sdk-go/api/instance/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+const (
+	instanceLabelPrefix = metaLabelPrefix + "instance_"
+
+	instanceNameLabel           = instanceLabelPrefix + "name"
+	instanceImageNameLabel      = instanceLabelPrefix + "image_name"
+	instanceStateLabel          = instanceLabelPrefix + "status"
+	instanceIDLabel             = instanceLabelPrefix + "id"
+	instanceZoneLabel           = instanceLabelPrefix + "zone"
+	instanceCommercialTypeLabel = instanceLabelPrefix + "commercial_type"
+	instanceTagsLabel           = instanceLabelPrefix + "tags"
+	instancePrivateIPLabel      = instanceLabelPrefix + "private_ip"
+	instancePublicIPLabel       = instanceLabelPrefix + "public_ip"
```

Comment:
maybe keep the same notation for ipv4 and ipv6?

Reply:
Done and I also use the same across baremetal and instance